crypto/cms/cms_env.c

```diff
@@ -198,8 +198,10 @@ static int cms_RecipientInfo_ktri_init(CMS_RecipientInfo *ri, X509 *recip,
     if (!cms_set1_SignerIdentifier(ktri->rid, recip, idtype))
         return 0;
 
-    X509_up_ref(recip);
-    EVP_PKEY_up_ref(pk);
+    if (!X509_up_ref(recip))
+        return 0;
+    if (!EVP_PKEY_up_ref(pk))
+        return 0;
```

Comment:
I think the error handling is not correct. I expect that when this function returns failure that the effect is that no up_ref calls happened, including the other return 0 calls below.

Reply:
Yeah, absolutely